plugins/CoreUpdater/Updater.php

```diff
@@ -124,15 +126,20 @@ public function updatePiwik($https = true)
             throw new UpdaterException($e, $messages);
         }
 
-        $partTwoUrl = Url::getCurrentUrlWithoutQueryString() . Url::getCurrentQueryStringWithParametersModified([
-            'action' => 'oneClickUpdatePartTwo',
-        ]);
+        $validFor10Minutes = time() + (60 * 10);
+        $nonce = Common::generateUniqId();
+        Option::set('NonceOneClickUpdatePartTwo', json_encode(['nonce' => $nonce, 'ttl' => $validFor10Minutes]));
 
-        $response = Http::sendHttpRequest($partTwoUrl, 300);
-        $response = @json_decode($response, $assoc = true);
+        PiwikCache::flushAll();
 
-        if (!empty($response)) {
-            $messages = array_merge($messages, $response);
+        $cliMulti = new CliMulti();
```

Comment:
not 100% sure it's a good idea using CliMulti but reckon it might be beneficial in case there Matomo can't be easily reached through the internet. This way it might cause problems only for those where async using CLI doesn't work and the Matomo can't be reached through the internet.

Reply:
eg in my case the original http call wouldn't have worked because:

> Got invalid response from API request: https://.... Response was 'curl_exec: SSL certificate problem: unable to get local issuer certificate. Hostname requested was: ...'